libs/admin-pages/src/pages/remove-list-confirmation/index.ts

```diff
@@ -0,0 +1,181 @@
+    artefactData,
+    hideLanguageToggle: true
+  });
+};
+
+const postHandler = async (req: Request, res: Response) => {
+  const lang = req.query.lng === "cy" ? cy : en;
+  const locale = req.query.lng === "cy" ? "cy" : "en";
+
+  const sessionData = req.session.removalData;
+  if (!sessionData || !sessionData.locationId || !sessionData.selectedArtefacts || sessionData.selectedArtefacts.length === 0) {
+    const lng = req.query.lng === "cy" ? "?lng=cy" : "";
+    return res.redirect(`/remove-list-search${lng}`);
+  }
+
+  const confirmation = req.body.confirmation;
+
+  if (!confirmation) {
+    const artefacts = await getArtefactsByIds(sessionData.selectedArtefacts);
+    const artefactData = artefacts.map((artefact) => {
+      const listType = mockListTypes.find((lt) => lt.id === artefact.listTypeId);
+      const listTypeName = listType ? (locale === "cy" ? listType.welshFriendlyName : listType.englishFriendlyName) : String(artefact.listTypeId);
+
+      const location = getLocationById(Number.parseInt(artefact.locationId, 10));
+      const courtName = location ? (locale === "cy" ? location.welshName : location.name) : artefact.locationId;
+
+      const displayDates = `${formatDateString(artefact.displayFrom)} to ${formatDateString(artefact.displayTo)}`;
+      const language = capitalizeFirstLetter(artefact.language);
+      const sensitivity = capitalizeFirstLetter(artefact.sensitivity);
+
+      return {
+        listType: listTypeName,
+        court: courtName,
+        contentDate: formatDateString(artefact.contentDate),
+        displayDates,
+        language,
+        sensitivity
+      };
+    });
+
+    return res.render("remove-list-confirmation/index", {
+      pageTitle: lang.pageTitle,
+      heading: lang.heading,
+      tableHeaders: lang.tableHeaders,
+      radioYes: lang.radioYes,
+      radioNo: lang.radioNo,
+      continueButton: lang.continueButton,
+      artefactData,
+      errors: [
+        {
+          text: lang.errorNoSelection,
+          href: "#confirmation"
+        }
+      ],
+      errorSummaryTitle: lang.errorSummaryTitle,
+      hideLanguageToggle: true
+    });
+  }
+
+  if (confirmation === "no") {
+    const lng = req.query.lng === "cy" ? "?lng=cy" : "";
+    return res.redirect(`/remove-list-search-results${lng}`);
+  }
+
+  try {
+    await deleteArtefacts(sessionData.selectedArtefacts);
+
+    delete req.session.removalData;
+    req.session.removalSuccess = true;
+
+    await new Promise<void>((resolve, reject) => {
+      req.session.save((err: Error | null | undefined) => {
+        if (err) reject(err);
+        else resolve();
+      });
+    });
+
+    const lng = req.query.lng === "cy" ? "?lng=cy" : "";
+    res.redirect(`/remove-list-success${lng}`);
+  } catch (error) {
+    console.error("Error deleting artefacts:", error);
+
+    const artefacts = await getArtefactsByIds(sessionData.selectedArtefacts);
+    const artefactData = artefacts.map((artefact) => {
```

Comment:
Any difference between lines 139-158 and lines 75-94. If not can it be moved up to a private method?